From 057bd4ec90ca3d4291cb4707d4ac95c8941569ef Mon Sep 17 00:00:00 2001
From: Dan Liang <dan.liang@intel.com>
Date: Thu, 2 May 2013 17:05:42 +0800
Subject: [PORT FROM MAIN] libstagefright: enhance error handler in MPEG4EXtractor

BZ: 92698

enhance error handler in MPEG4Extractor, there's case that nalLength > size
but nalLength + 4 < size, for ex nalLength = 0xffffffff.

Change-Id: I6036d34261ca8591679082c3a85c7e34ee42a393
Orig-Change-Id: I7d950a6a5a6d24bad68c9472f24313206c4a796d
Signed-off-by: ywan171 <yi.a.wang@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/MPEG4Extractor.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index ad38a41..987ce13 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -3279,7 +3279,7 @@ status_t MPEG4Source::read(
                 if (!isMalFormed) {
                     nalLength = parseNALSize(&mSrcBuffer[srcOffset]);
                     srcOffset += mNALLengthSize;
-                    isMalFormed = srcOffset + nalLength > size;
+                    isMalFormed = nalLength > size | srcOffset + nalLength > size;
                 }
 
                 if (isMalFormed) {
-- 
1.7.4.1

